CarpetX/src/valid.cxx

```diff
@@ -229,24 +229,28 @@ void poison_invalid_ga(const int gi, const int vi, const int tl) {
 
   auto &restrict globaldata = ghext->globaldata;
   auto &restrict arraygroupdata = *globaldata.arraygroupdata.at(gi);
+  cGroup group;
+  int ierr = CCTK_GroupData(gi, &group);
+  assert(!ierr);
 
   const valid_t &valid = arraygroupdata.valid.at(tl).at(vi).get();
   if (valid.valid_all())
     return;
 
-  CCTK_REAL poison;
-  std::memcpy(&poison, &ipoison, sizeof poison);
+  const size_t typesize = size_t(CCTK_VarTypeSize(group.vartype));
+  char poison[typesize];
```

Comment:
Didn't you switch to `std::vector` elsewhere for a similar construct?

Reply:
Yes, and *here* it dies not complain about the varuable sized array, but it diez in io_openpmd.cxx. I have no clue why.

I will redo thus all anyway, to have one poision code for bith places where it is used.